components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/PlacesBookmarksStorage.kt

```diff
@@ -79,11 +80,22 @@ open class PlacesBookmarksStorage(context: Context) : PlacesStorage(context), Bo
      * Retrieves a list of recently added bookmarks.
      *
      * @param limit The maximum number of entries to return.
+     * @param daysLimit Optional parameter to filter entries older than this number of days
      * @return The list of matching bookmark nodes up to the limit number of items.
      */
-    override suspend fun getRecentBookmarks(limit: Int): List<BookmarkNode> {
+    override suspend fun getRecentBookmarks(
+        limit: Int,
+        daysLimit: Long?
```

Comment:
I would use `maxAge` here in ms. Then the caller can specify 7 days or 4 hours etc. using `TimeUnit.DAYS.toMillis(7)` or `TimeUnit.HOURS.toMillis(4)`.

Reply:
I would have wanted to hide the implementation from the client, so it would not have to know what API to use in order to transform days in ms, but rather provide a simple number of days as a parameter.

Reply:
@mcarare Right, but this way you're limiting the API to days, which seems unneeded? And date/time conversion APIs are not something we should be hiding from the consumer. These are standard APIs and optional. I could just as well pass `7 * 24 * 60 * 60 * 1000`, which we do in most places actually :).

Reply:
@mcarare But if you like this much better we you could also have both. Then one method can call the other :)

Reply:
No, I was just explaining the reason for my first choice. But you are right, maybe this could be used with other units like  hours, and we should not limit usage to a specific unit but rather provide the smallest one available (milliseconds)

Reply:
OK, looks good to me! Let's just add the test and land this :).